qiskit_metal/designs/design_base.py

```diff
@@ -447,37 +420,45 @@ def rebuild(self):  # remake_all_components
         for _, obj in self._components.items():  # pylint: disable=unused-variable
             obj.rebuild()
 
-    def reload_component(self, component_module_name: str,
-                         component_class_name: str):
-        """Reload the module and class of a given component and update all
-        class instances. (Advanced function.)
+    def reload_and_rebuild_components(self, qis_abs_path: str):
+        """
+        Reload the module and class of a given component and updates
+        all class instances. Then rebuilds all QComponents of that class
 
         Arguments:
-            component_module_name (str): String name of the module name, such as
-                `qiskit_metal.qlibrary.qubits.transmon_pocket`
-            component_class_name (str): String name of the class name inside thst module,
-                such  as `TransmonPocket`
+            qis_abs_path: Absolute to the QComponent source file to be reloaded
+
+        Raises:
+            QiskitMetalDesignError: The given name is a magic method not in the dictionary
         """
-        self.logger.debug(
-            f'Reloading component_class_name={component_class_name}; component_module_name={component_module_name}'
-        )
-        module = importlib.import_module(component_module_name)
-        module = importlib.reload(module)
-        new_class = getattr(module, component_class_name)
-
-        # components that need
-        for instance in filter(
-                lambda k: k.__class__.__name__ == component_class_name and k.
-                __class__.__module__ == component_module_name,
-                self._components.values()):
-            instance.__class__ = new_class
-
-        # Alternative, but reload will say not in sys.path
-        # self = gui.component_window.src_widgets[-1].ui.src_editor
-        # spec = importlib.util.spec_from_file_location(self.component_module_name, self.component_module_path) # type: ModuleSpec
-        # module = importlib.util.module_from_spec(spec) # type: module e.g.,
-        # spec.loader.exec_module(module)
-        # importlib.reload(module)
+
+        try:
+            for comp in self.components:  # runs so few times, it's fine to reload object multiple times
```

Comment:
The intentions is to add 100,000+ qcomponents in a qdesign.  Why is this code doing this?

Reply:
2 reasons.
The first is that this mode is for developing an individual QComponent so we don't anticipate anyone having more than 1-5 QComponents in their circuit.

The second reason is that this is part of the code while will get overhauled in the next PR.  Because we're no longer sending individual filenames to the design_base, we actually need a clean way for design_base to read the `dirtied_files` from the File Watcher. 
Currently the File Watcher is part of the Model for the QLibrary display, but in the next PR, that is getting pulled out and made into a separate piece, at which point I'll reimplement the `design_base` logic to re-import any edited file instead of just the QComponents being used.